src/web/layouts/InteractiveImmersiveLayout.tsx

```diff
@@ -61,7 +61,10 @@ const Renderer: React.FC<{
 		});
 
 		return ok ? (
-			<figure id={'id' in element ? element.id : undefined} key={index}>
+			<figure
+				id={'elementId' in element ? element.elementId : undefined}
```

Comment:
Can we type the object this is coming from at all to try avoid these things from happening in general, e.g. if we had `element` and all its fields typed in an interface, we'd have caught this sooner

Reply:
Yeah, we could switch on type. I think I did it this way originally as it would be quite verbose to handle it otherwise but it would be safer.